src/prefect/server/api/server.py

```diff
@@ -801,52 +801,32 @@ def is_port_available(port: int):
     def address(self) -> str:
         return f"http://127.0.0.1:{self.port}"
 
-    def start(self):
+    def start(self, timeout: Optional[int] = None):
         """
         Start the server in a separate process. Safe to call multiple times; only starts
         the server once.
+
+        Args:
+            timeout: The maximum time to wait for the server to start
         """
         if not self.running:
             subprocess_server_logger.info(f"Starting server on {self.address()}")
             try:
                 self.running = True
-                server_env = {"PREFECT_UI_ENABLED": "0"}
-                self.server_process = subprocess.Popen(
-                    args=[
-                        get_sys_executable(),
-                        "-m",
-                        "uvicorn",
-                        "--app-dir",
-                        # quote wrapping needed for windows paths with spaces
-                        f'"{prefect.__module_path__.parent}"',
-                        "--factory",
-                        "prefect.server.api.server:create_app",
-                        "--host",
-                        "127.0.0.1",
-                        "--port",
-                        str(self.port),
-                        "--log-level",
-                        "error",
-                        "--lifespan",
-                        "on",
-                    ],
-                    env={
-                        **os.environ,
-                        **server_env,
-                        **get_current_settings().to_environment_variables(
-                            exclude_unset=True
-                        ),
-                    },
-                )
+                self.server_process = self._run_uvicorn_command()
                 atexit.register(self.stop)
-
                 with httpx.Client() as client:
                     response = None
                     elapsed_time = 0
-                    while (
-                        elapsed_time
-                        < PREFECT_SERVER_EPHEMERAL_STARTUP_TIMEOUT_SECONDS.value()
-                    ):
+                    max_wait_time = (
+                        timeout
+                        or PREFECT_SERVER_EPHEMERAL_STARTUP_TIMEOUT_SECONDS.value()
+                    )
+                    while elapsed_time < max_wait_time:
+                        if self.server_process.poll() == 3:
+                            self.port = self.find_available_port()
+                            self.server_process = self._run_uvicorn_command()
+                            continue
```

Comment:
When running the tests in CI, there was a race condition where the test server would fail to start because of port conflicts. This change will let the `SubprocessASGIServer` class grab a new port on port conflicts.